modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/api/AttributeDefinition.java

```diff
@@ -264,6 +268,20 @@ public static Builder fromChildListConfiguration(Class<?> type)
         }
 
         /**
+         * Used when the attribute type is collection and the collection implementation must
+         * be of a given type.
+         *
+         * @param type collection type.
+         * @return the builder.
+         */
+        public Builder withCollectionType(Class<? extends Collection> type)
```

Comment:
wouldn't it be better if you receive a Supplier instead so that you can fine tune how the collection is created?